### PR TITLE
MIgrate away from Kotlin Documentable

### PR DESCRIPTION
**Pull request by romanowski**
Scala3doc was only loosly using dokka model and it has started to blocking us.

This PR standarize over Member and expand semantic of Kind. This allow us to have more typesafe way to reason about code.

It will also make migration away from dokka easier if needed.

This PR contains #10702 that allow to link to external APIs as well as refactor linking.

Beside that this PR document correctly all members as well as allows to link to given ones.

Signature are rendered correctly (no problem with annotations etc.)

### CODE REVIEW

File: scala3doc/src/dotty/dokka/compat.scala
Comment (abgruszecki) [inline scala3doc/src/dotty/dokka/compat.scala:31]: Can we preemptively use `Collections.emptyList` here? I'd prefer to avoid issues due to some code modifying our "empty" list.

File: scala3doc/src/dotty/dokka/compat.scala
Comment (abgruszecki) [inline scala3doc/src/dotty/dokka/compat.scala:34]: Ditto.

**Review by abgruszecki — COMMENTED** [review]
There's a whole bunch of changes in code areas I'm not familiar with. I don't have the time to try and understand them now, someone else should review this as well.

File: scala3doc/src/dotty/dokka/translators/ScalaPageCreator.scala
Comment (BarkingBad) [inline scala3doc/src/dotty/dokka/translators/ScalaPageCreator.scala:51]: Maybe we should add more explanatory message why calling this method fails than "Not Implemented"

File: scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala
Comment (BarkingBad) [inline scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala:30]: Too deep indentation?

File: scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala
Comment (BarkingBad) [inline scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala:433]: I know we will discuss whether dokka will still remain as a base for our doctool, but if dokka stays, we should consider making `Member` a subtype od `Documentable`, and get rid of these hacks with `DClass` wrapper :P

**Review by BarkingBad — COMMENTED** [review]
There are some indentations mistakes in `ClassLikeSupport`. Otherwise, looks Ok to me

File: scala3doc/src/dotty/dokka/model/api/api.scala
Comment (romanowski) [reply scala3doc/src/dotty/dokka/model/api/api.scala:105]: variance goes before name and in some cases we would like to refer only to name without variance

File: scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala
Comment (romanowski) [reply scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala:449]: Actually using DClass is really conviniet :)

File: scala3doc/src/dotty/dokka/translators/ScalaSignatureProvider.scala
Comment (pikinier20) [inline scala3doc/src/dotty/dokka/translators/ScalaSignatureProvider.scala:52]: Member?

Comment (romanowski) [reply scala3doc/src/dotty/dokka/translators/ScalaSignatureProvider.scala:52]: I will get rid of all `Documentable` after this PR is merged

**Review by abgruszecki — APPROVED** [review]

**Pull request merged**
